feature/dashboard/src/main/java/com/simprints/feature/dashboard/settings/syncinfo/SyncInfoViewModel.kt

```diff
@@ -50,193 +57,358 @@ internal class SyncInfoViewModel @Inject constructor(
-
-    val recordsToDownSync: LiveData<DownSyncCounts?>
-        get() = _recordsToDownSync
-    private val _recordsToDownSync = MutableLiveData<DownSyncCounts?>(null)
-
-    val moduleCounts: LiveData<List<ModuleCount>>
-        get() = _moduleCounts
-    private val _moduleCounts = MutableLiveData<List<ModuleCount>>()
-
-    val configuration: LiveData<ProjectConfiguration>
-        get() = _configuration
-    private val _configuration = MutableLiveData<ProjectConfiguration>()
-
-    private val isConnected: LiveData<Boolean> = connectivityTracker.observeIsConnected()
-
-    val lastSyncState = eventSyncManager.getLastSyncState()
-    private var lastKnownEventSyncState: EventSyncState? = null
-
-    val isSyncAvailable: LiveData<Boolean>
-        get() = _isSyncAvailable
-    private val _isSyncAvailable = MediatorLiveData<Boolean>()
-
-    val isReloginRequired: LiveData<Boolean>
-        get() = _isReloginRequired
-    private val _isReloginRequired = MediatorLiveData<Boolean>()
-
-    val loginRequestedEventLiveData: LiveData<LiveDataEventWithContent<Bundle>>
-        get() = _loginRequestedEventLiveData
-    private val _loginRequestedEventLiveData = MutableLiveData<LiveDataEventWithContent<Bundle>>()
-
-    init {
-        _isSyncAvailable.addSource(lastSyncState) { lastSyncStateValue ->
-            _isSyncAvailable.postValue(
-                emitSyncAvailable(
-                    isSyncRunning = lastSyncStateValue?.isSyncRunning(),
-                    isConnected = isConnected.value,
-                    syncConfiguration = configuration.value?.synchronization,
-                ),
+    var isPreLogoutUpSync = false
+
+    val loginNavigationEventLiveData: LiveData<LoginParams>
+        get() = _loginNavigationEventLiveData
+    private val _loginNavigationEventLiveData = MutableLiveData<LoginParams>()
+
+    private val eventSyncStateFlow =
+        eventSyncManager.getLastSyncState(useDefaultValue = true /* otherwise value not guaranteed */).asFlow()
+    private val imageSyncStatusFlow =
+        syncOrchestrator.observeImageSyncStatus()
+
+    val logoutEventLiveData: LiveData<Unit?> = combine(
+        eventSyncStateFlow,
+        imageSyncStatusFlow,
+    ) { eventSyncState, imageSyncStatus ->
+        val isReadyToLogOut =
+            isPreLogoutUpSync && eventSyncState.isSyncCompleted() && !imageSyncStatus.isSyncing
+                && !configManager.isModuleSelectionRequired()
+        return@combine isReadyToLogOut
+    }.debounce(LOGOUT_DELAY_MILLIS).transformLatest { isReadyToLogOut ->
+        if (isReadyToLogOut) {
+            // "flick" the logout event to prevent persistence while not observed, and to avoid unexpected logout later
+            emit(Unit)
+            emit(null)
+        }
+    }.asLiveData()
+
+    val syncInfoLiveData: LiveData<SyncInfo> = combine8(
+        connectivityTracker.observeIsConnected().asFlow(),
+        authStore.watchSignedInProjectId().map(String::isNotEmpty),
```

Comment:
In relation to all the new watch* functions - observe is more idiomatic to Kotlin/Android and I would prefer to stick to that.